### PR TITLE
issue: 1041212 Fixed unexpected behavior UDP_MAP_REMOVE

### PR DESCRIPTION
**Pull request by serglelv**
Signed-off-by: Sergey Lykov <sergeyly@mellanox.com>

### CODE REVIEW

**Comment by serglelv** [comment]
Please review.
@olgk

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1517/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
The fix was discussed with Oleg and it seems an issue in the test.  The PR should be closed without merging.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1587/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
The fix was updated. Please review.
@olgk @ilansmith

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1588/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
The commit message was updated.
@ilansmith @olgk

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1595/ for details (Mellanox internal link).

**Comment by serglelv** [comment]
The fix was simplified. Based on Liran's feedback.  Please review.
@olgk @liranoz12

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1635/ for details (Mellanox internal link).

File: src/vma/sock/sockinfo_udp.cpp [hunk 1]
```diff
@@ -1231,7 +1231,11 @@ int sockinfo_udp::setsockopt(int __level, int __optname, __const void *__optval,
 			si_udp_logdbg("stopping de-muxing packets to port %d", ntohs(port));
 			m_port_map_lock.lock();
 			m_port_map.erase(std::remove(m_port_map.begin(), m_port_map.end(), port), m_port_map.end());
-			m_sockopt_mapped = false;
+			if (m_port_map.size()) {
+				m_sockopt_mapped = true;
+			} else {
+				m_sockopt_mapped = false;
+			}
```
Comment (liranoz12) [inline src/vma/sock/sockinfo_udp.cpp]: Hi,
There is not reason to change _m_sockopt_mapped_ to true while removing an item from the map.
Additionally, we should call _set_rx_packet_processor()_ only when we erase the last element from the map (similar approach should also be applied to UDP_MAP_ADD).
Please see patch suggestion at redmine.

Comment (olgk) [reply src/vma/sock/sockinfo_udp.cpp]: Agree with Liran's comment, Sergey, please check suggested patch.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1639/ for details (Mellanox internal link).

**Comment by liranoz12** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1641/ for details (Mellanox internal link).

**Comment by liranoz12** [comment]
Merged #435

**Pull request closed without merging**